src/webapp/pages/app/App.tsx

```diff
@@ -17,36 +17,36 @@ import "./App.css";
 import muiThemeLegacy from "./themes/dhis2-legacy.theme";
 import { muiTheme } from "./themes/dhis2.theme";
 import _ from "lodash";
+import { CustomHeader } from "../../components/custom-header/CustomHeader";
+import { CustomFooter } from "../../components/custom-footer/CustomFooter";
 
 export interface AppProps {
     api: D2Api;
     d2: D2;
     instance: Instance;
 }
 
-function getSettings(settingsFromStorage: Maybe<Settings>, defaultSettings: Settings): Settings {
-    if (!settingsFromStorage) throw new Error("Cannot load settings");
-    const hasTemplates = settingsFromStorage ? settingsFromStorage.templates.length > 0 : false;
-    const settings = hasTemplates ? settingsFromStorage : defaultSettings;
-    return settings;
-}
-
 export const App: React.FC<AppProps> = React.memo(function App({ api, d2, instance }) {
     const [loading, setLoading] = useState(true);
     const [appContext, setAppContext] = useState<AppContextState | null>(null);
 
     useEffect(() => {
         async function setup() {
             const isDev = process.env.NODE_ENV === "development";
-            const defaultSettings = await fetch("default-settings.json").then<Settings>(res => res.json());
             const storageName = (process.env.REACT_APP_STORAGE as Maybe<StorageName>) || "datastore";
             const compositionRoot = getCompositionRoot(api, instance, storageName);
             const currentUser = (await compositionRoot.users.getCurrent.execute().runAsync()).data;
             const settingsFromStorage = (await compositionRoot.settings.get.execute().runAsync()).data;
-            const settings = getSettings(settingsFromStorage, defaultSettings);
             const pluginVersion = `${_.get(d2, "system.version.major")}${_.get(d2, "system.version.minor")}`;
-            if (!settings.id) {
-                await compositionRoot.settings.save.execute(settings).runAsync();
+            let settings: Settings;
```

Comment:
Not a big deal here, but as a rule we prefer immutable variables. Can we move all the logic to a function to avoid mutating the settings variable?

Something like:

```ts
async function getSettingsOrInitialize(
    settingsFromStorage: Maybe<Settings>,
    compositionRoot: CompositionRoot
): Promise<Settings> {
    // logic here
    return settings;
}
```